cpp/test/sg/genetic/program_test.cu

```diff
@@ -38,7 +38,10 @@ namespace genetic {
 class GeneticProgramTest : public ::testing::Test {
  public:
   GeneticProgramTest()
-    : d_data(0, cudaStream_t(0)),
+    : nodes1_buffer(7 * sizeof(node), stream),
```

Comment:
By constructing device_buffer in the constructor of the fixture, we can eliminate the raw allocation and deallocation in SetUp and TearDown.